skglm/solvers/cd_solver.py

```diff
@@ -226,32 +221,41 @@ def cd_solver(
     obj_out = []
     all_feats = np.arange(n_features)
     stop_crit = np.inf  # initialize for case n_iter=0
-    w_acc, Xw_acc = np.zeros(n_features), np.zeros(n_samples)
+    w_acc, Xw_acc = np.zeros(n_features + fit_intercept), np.zeros(n_samples)
 
     is_sparse = sparse.issparse(X)
+
+    if len(w) != n_features + fit_intercept:
+        raise ValueError(
+            "The size of coefficients should be n_features + fit_intercept, \
```

Comment:
beware, this renders as:
`ValueError: The size of coefficients should be n_features + fit_intercept,                 expected 61, got 60`